### ISSUE TITLE
Add support for remappings files

### ISSUE THREAD

**Issue by magnetto90** [crytic/medusa#207]
It would be nice to be able to use remappings file from Foundry since Medusa is more powerful, I think is a good idea to have an easy way to replace Foundry fuzzer with Medusa within a Foundry project.

**Comment by anishnaik** [comment]
Hey @magnetto90 , the underlying compilation platform we use, `crytic-compile`, should use foundry under-the-hood and handle any remappings. Is there a repo in question that you are having trouble with? If your `foundry.toml` file has the remappings in there, then `medusa` should work as expected.

**Comment by magnetto90** [comment]
@anishnaik thanks for replying, I was trying to use it in this auditing contest: https://github.com/Cyfrin/2023-08-sparkn and I was having compilation errors. Let me try again and see if I can share my problem.

**Comment by magnetto90** [comment]
`medusa.json` is in the root folder and is configured as we did in the workshop and my test case is in a folder called "medusa" also in the root of the project.

### Version: `medusa version 0.1.1`

[Link to Distributor.sol](https://github.com/Cyfrin/2023-08-sparkn/blob/main/src/Distributor.sol)

[Link to foundry.toml](https://github.com/Cyfrin/2023-08-sparkn/blob/main/foundry.toml)

## My test contract

```solidity
// SPDX-License-Identifier: MIT
pragma solidity 0.8.18;

import "../src/Distributor.sol";

// Run with medusa fuzz --target medusa/DistributorTest.sol --deployment-order DistributorTest

contract DistributorTest is Distributor(address(1), address(2)) {
    function fuzz_getConstants() public returns (bool) {
        (,,, uint8 version) = getConstants();

        return version == 1;
    }
}
```

## Error Log

```
➜  2023-08-sparkn git:(medusa2) ✗ medusa fuzz --target medusa/DistributorTest.sol --deployment-order DistributorTest
⇾ Reading the configuration file at: /home/audits/2023-08-sparkn/medusa.json
⇾ Compiling targets with crytic-compile
error Failed to compile target error="error while executing crytic-compile:\nOUTPUT:\nINFO:CryticCompile:Compilation warnings/errors on medusa/DistributorTest.sol:\nError: Source \"openzeppelin/token/ERC20/IERC20.sol\" not found: File not found. Searched the following locations: \"\".\n  --> src/Distributor.sol:26:1:\n   |\n26 | import {IERC20} from \"openzeppelin/token/ERC20/IERC20.sol\";\n   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n\nError: Source \"openzeppelin/token/ERC20/utils/SafeERC20.sol\" not found: File not found. Searched the following locations: \"\".\n  --> src/Distributor.sol:27:1:\n   |\n27 | import {SafeERC20} from \"openzeppelin/token/ERC20/utils/SafeERC20.sol\";\n   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n\nError: Source \"openzeppelin/access/Ownable.sol\" not found: File not found. Searched the following locations: \"\".\n  --> src/ProxyFactory.sol:26:1:\n   |\n26 | import {Ownable} from \"openzeppelin/access/Ownable.sol\";\n   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n\nError: Source \"openzeppelin/utils/cryptography/ECDSA.sol\" not found: File not found. Searched the following locations: \"\".\n  --> src/ProxyFactory.sol:27:1:\n   |\n27 | import {ECDSA} from \"openzeppelin/utils/cryptography/ECDSA.sol\";\n   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n\nError: Source \"openzeppelin/utils/cryptography/EIP712.sol\" not found: File not found. Searched the following locations: \"\".\n  --> src/ProxyFactory.sol:28:1:\n   |\n28 | import {EIP712} from \"openzeppelin/utils/cryptography/EIP712.sol\";\n   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n\n\nERROR:CryticCompile:Invalid solc compilation Error: Source \"openzeppelin/token/ERC20/IERC20.sol\" not found: File not found. Searched the following locations: \"\".\n  --> src/Distributor.sol:26:1:\n   |\n26 | import {IERC20} from \"openzeppelin/token/ERC20/IERC20.sol\";\n   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n\nError: Source \"openzeppelin/token/ERC20/utils/SafeERC20.sol\" not found: File not found. Searched the following locations: \"\".\n  --> src/Distributor.sol:27:1:\n   |\n27 | import {SafeERC20} from \"openzeppelin/token/ERC20/utils/SafeERC20.sol\";\n   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n\nError: Source \"openzeppelin/access/Ownable.sol\" not found: File not found. Searched the following locations: \"\".\n  --> src/ProxyFactory.sol:26:1:\n   |\n26 | import {Ownable} from \"openzeppelin/access/Ownable.sol\";\n   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n\nError: Source \"openzeppelin/utils/cryptography/ECDSA.sol\" not found: File not found. Searched the following locations: \"\".\n  --> src/ProxyFactory.sol:27:1:\n   |\n27 | import {ECDSA} from \"openzeppelin/utils/cryptography/ECDSA.sol\";\n   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n\nError: Source \"openzeppelin/utils/cryptography/EIP712.sol\" not found: File not found. Searched the following locations: \"\".\n  --> src/ProxyFactory.sol:28:1:\n   |\n28 | import {EIP712} from \"openzeppelin/utils/cryptography/EIP712.sol\";\n   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n\n\n\nERROR: exit status 255\n"
```

Please let me know I you need more info, I assume I have some error in my config.

**Comment by 0xalpharush** [comment]
These look like warnings from solc and not medusa/crytic-compile. Are you able to run `forge build` successfully? Based off the missing files being related openzeppelin, you may need to run `forge install`

**Comment by magnetto90** [comment]
@0xalpharush I am running `forge test` without problem

![image](https://github.com/crytic/medusa/assets/46317127/5c01a1f7-bff4-484e-aa9c-d13aaca76942)

**Comment by anishnaik** [comment]
Hey @magnetto90 let's do two things:
1. Can you run `medusa fuzz --target . --deployment-order DistributorTest`. What this does is it tells `crytic-compile` to compile _everything_ and then the actual "target contract" is `DistributorTest`. This is definitely a little unintuitive so we need to improve our internal documentation about it. More generally, when you have an actual "project" versus a single file (which was the case for the secureum workshop), 99.9% of the time you will want to run `medusa fuzz --target . etc etc`.
2. We have a new release of `medusa` if you want to try it out. The errors look a lot nicer and are easier on the eyes ;) It also has new features like `console.log` so you can easily log stuff in your fuzz tests.

**Comment by magnetto90** [comment]
1. I used the command but I having a different error. I will dig more into it.
```
⇾ Compiling targets with crytic-compile
error Failed to initialize the test chain error="DeploymentOrder specified a contract name which was not found in the compilation: DistributorTest\n"
```
2. Sure, I compiled the previous version, should I pull the main branch or the new version is in another branch?

**Comment by anishnaik** [comment]
1. Yea so my guess on this one is that the contracts in the `medusa/` folder will not be compiled. You should probably move to under `test` or `src`.
2. Yup you can pull from master and run `go build` and then move it to somewhere in your path. You can also download the binary here: https://github.com/crytic/medusa/releases/tag/v0.1.2

**Comment by magnetto90** [comment]
Now is working! Thanks for the help, also the error log in the new version looks much more clean! Thanks!

**Comment by anishnaik** [comment]
Awesome, glad it worked out :) Closing the issue
